content/local/store.go

```diff
@@ -687,15 +687,19 @@ func writeTimestampFile(p string, t time.Time) error {
 }
 
 func atomicWrite(path string, data []byte, mode os.FileMode) error {
-	tmp := fmt.Sprintf("%s.tmp", path)
-	f, err := os.OpenFile(tmp, os.O_RDWR|os.O_CREATE|os.O_TRUNC|os.O_SYNC, mode)
+	f, err := ioutil.TempFile(filepath.Dir(path), ".tmp-"+filepath.Base(path))
```

Comment:
> Should be changed to write a random temporary file Instead of writing a fixed temporary file.

The path will be locked by content store. In content store, I think it is safe to use `%s.tmp` in here.

Anyway, I am ok to align with `continuity` package. :) And why not just use the existing function?

Reply:
If it does not concurrent writing of `%s.tmp`, then there is actually no need to atomic write, right?
I found that it is to fix some failures in Windows CI  #3324
However, this modification cannot guarantee Linux atomic writing, nor does it guarantee windows atomic writing.
It's just a temporary solution that can pass CI.
I think it is necessary to implement the atomic write of Windows and then use it here

Reply:
The atomic write is to make sure that the content is completeness. Sometimes, the disk has pressure and the content will be truncated somehow. The atomic write is like transaction, not just for concurrent.

Reply:
Thank you, I understand, i found the original PR #2580.
it was forcibly interrupted when the file was just opened and no data was written. error caused by empty data when reading.
Although I think we should deal with empty data in this case, instead of guaranteeing to write data here, but thank you.

Reply:
The atomicWrite is used to ensure the content should be completeness, not empty or part of it. :)